src/gui/editors/PianoRoll.cpp

```diff
@@ -4357,15 +4357,17 @@ void PianoRollWindow::stop()
 
 void PianoRollWindow::record()
 {
-	m_editor->record();
+	Engine::getSong()->startRecordCountDown();
+	QTimer::singleShot( Engine::getSong()->getRecordDelay() * 1000 , m_editor, SLOT( record() ) );
```

Comment:
This is because I haven't familiar with the rendering process of LMMS, I will try to understand it before doing this.